taipy/core/taipy.py

```diff
@@ -873,6 +873,23 @@ def get_cycles() -> List[Cycle]:
     return _CycleManagerFactory._build_manager()._get_all()
 
 
+def can_create(config: Union[ScenarioConfig, DataNodeConfig]) -> Reasons:
+    """Indicate if a config can be created.
+
+    This function checks if the given scenario or data node config can be created.
+
+    Returns:
+        True if the given config can be created. False otherwise.
+    """
+    if isinstance(config, ScenarioConfig):
+        return _ScenarioManagerFactory._build_manager()._can_create(config)
+    elif isinstance(config, DataNodeConfig):
+        return _DataManagerFactory._build_manager()._can_create(config)
+    else:
+        config_id = getattr(config, "id", None) or str(config)
+        return Reasons(config_id)._add_reason(config_id, _build_config_can_not_create_reason(config_id))
```

Comment:
If it is not a `ScenarioConfig` or a `DataNodeConfig`, the reason to return is that `config` has an incorrect type. There is no need to look for a `config_id`. No?

```suggestion
        return Reasons(str(config))._add_reason(str(config), _build_config_can_not_create_reason(config_id))
```

What do you think?

Reply:
If it's a Section object, `str(config)` will return its id?

We need the id here right?

Reply:
My point is that if the `config` variable is not a `ScenarioConfig `or a `DataNodeConfig`, it means the user created a bug somehow. It can be anything, like an integer, a list, or a more complex object like a Dataframe, or whatever. It looks strange to me to assume that it is maybe a `Section`, and so I don't get the point of trying to get an `id` from it. We just want a way to have a text representation of the object to let the end-user know it has an incorrect type.

But as I said on another comment, this is a minor remark. I am fine with your proposal.